src/Algebra/Graph/Label.hs

```diff
@@ -281,11 +282,17 @@ fromExtended :: Extended a -> Maybe a
 fromExtended (Finite a) = Just a
 fromExtended Infinite   = Nothing
 
-instance Num a => Num (Extended a) where
+-- A type alias for a binary function on Extended.
+type ExtendedBinary a = Extended a -> Extended a -> Extended a
```

Comment:
Looks like this type is used only in the definition of `(*)` -- I suggest simply inlining this definition.